File: data_structures/q4/solution.cpp
#include<iostream>
#include<vector>
using namespace std;

vector<int> STACK;

void push(int x){
    STACK.push_back(x);
}
void pop(){
    STACK.pop_back();
}
int peek(){
    return STACK[-1];
}

int main(){
    push(69);
    push(29);
    cout << "[";
    for(int element : STACK){
        cout << "\t" << element << " , ";
    } 
    cout << "]" << endl;
    pop();
    peek();
    cout << "[";
    for(int element : STACK){
        cout << "\t" << element << "";
    }
    cout << "]" << endl;
    return 0;
}

File: data_structures/q4/solution.py
STACK_LIST = []
# push, pop, peek
def push(n):
    STACK_LIST.append(n)

def pop():
    STACK_LIST.pop()

def peek():
    return STACK_LIST[-1]

print(STACK_LIST)
push(34)
push(2)
print(STACK_LIST)
pop()
print(STACK_LIST)
print(peek())
print(STACK_LIST)




